### ISSUE TITLE
Regularisations for training

### ISSUE THREAD

**Issue by mdrumond** [mdrumond/coltrain#13]
These are needed for fancy imagenet training

- [ ] Dropout
- [ ] Batch normalization
- [ ] L2 Regularizations
